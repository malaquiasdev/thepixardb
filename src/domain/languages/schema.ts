export const schema = {
  tags: ['languages'],
  description: 'Our languages are standard by ISO 639-1. Default value is en-US.',
  response: {
    200: {
      type: 'array',
      items: {
        type: 'object',
        properties: {
          iso6391: { type: 'string' },
          description: { type: 'string' },
        },
      },
    },
  },
};
