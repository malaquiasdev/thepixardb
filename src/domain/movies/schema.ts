export const moviesSchema = {
  tags: ['movies'],
  description: 'Get the primary information about a movie.',
  querystring: {
    type: 'object',
    additionalProperties: false,
    properties: {
      language: {
        type: 'string',
        description: 'Pass a ISO 639-1 value to display translated data for the fields that support it.',
        default: 'en',
        enum: ['en', 'pt'],
      },
      genre: {
        type: 'string',
      },
    },
  },
  response: {
    200: {
      type: 'array',
      items: {
        type: 'object',
        properties: {
          id: { type: 'string' },
          title: { type: 'string' },
          description: { type: 'string' },
          releaseYear: { type: 'string' },
          runtime: { type: 'number' },
          posterUrl: { type: 'string' },
          bannerUrl: { type: 'string' },
        },
      },
    },
  },
};

export const movieSchema = {
  tags: ['movies'],
  description: '',
  params: {
    type: 'object',
    properties: {
      id: {
        type: 'string',
        description: 'movie id',
      },
    },
  },
  querystring: {
    type: 'object',
    additionalProperties: false,
    properties: {
      language: {
        type: 'string',
        description: 'Pass a ISO 639-1 value to display translated data for the fields that support it.',
        default: 'en',
        enum: ['en', 'pt'],
      },
    },
  },
  response: {
    200: {
      type: 'object',
      properties: {
        id: { type: 'string' },
        title: { type: 'string' },
        description: { type: 'string' },
        releaseYear: { type: 'string' },
        runtime: { type: 'number' },
        poster: {
          type: 'object',
          properties: {
            width: { type: 'integer' },
            height: { type: 'integer' },
            url: { type: 'string' },
          },
        },
      },
    },
  },
};
