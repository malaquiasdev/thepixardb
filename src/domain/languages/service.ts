export function listAllowedLanguages() {
  return [
    {
      iso6391: 'JA',
      description: 'Japanese',
    },
    {
      iso6391: 'EN_US',
      description: 'English (United States)',
    },
    {
      iso6391: 'PT_BR',
      description: 'Portuguese (Brazil)',
    },
  ];
}
