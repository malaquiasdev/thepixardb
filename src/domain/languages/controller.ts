import { FastifyReply } from 'fastify';
import { listAllowedLanguages } from './service';

const SEVEN_DAYS_IN_SECONDS = 604800;

export async function languageController(_, reply: FastifyReply) {
  return reply
    .headers({
      'Cache-Control': `max-age=${SEVEN_DAYS_IN_SECONDS}`,
      'Access-Control-Allow-Origin': '*',
    })
    .send(listAllowedLanguages());
}
