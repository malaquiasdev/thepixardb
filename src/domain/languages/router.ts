import { languageController } from './controller';
import { schema } from './schema';

export default function languageRouter(server, _, next) {
  server.route({
    url: '/languages',
    logLevel: 'warn',
    method: ['GET', 'HEAD', 'OPTIONS'],
    handler: languageController,
    schema,
  });
  next();
}
