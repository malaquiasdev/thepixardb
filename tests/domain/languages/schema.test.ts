import { schema } from '../../../src/domain/languages/schema';

describe('schema', () => {
  it('should return an object', () => {
    expect(schema).toBeInstanceOf(Object);
  });
  it('should match with the snapshot', () => {
    expect(schema).toMatchSnapshot();
  });
});
