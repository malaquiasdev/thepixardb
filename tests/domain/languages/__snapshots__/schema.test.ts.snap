// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema should match with the snapshot 1`] = `
Object {
  "description": "Our languages are standard by ISO 639-1. Default value is en-US.",
  "response": Object {
    "200": Object {
      "items": Object {
        "properties": Object {
          "description": Object {
            "type": "string",
          },
          "iso6391": Object {
            "type": "string",
          },
        },
        "type": "object",
      },
      "type": "array",
    },
  },
  "tags": Array [
    "languages",
  ],
}
`;
